### ISSUE TITLE
Proton cant create new prefixes on Proton 8 and Experimental

### ISSUE THREAD

**Issue by jkrhu** [ValveSoftware/Proton#6859]
Hi!

For some reason Proton is unable to create new prefixes for installed games when using Proton 8 or Experimental. I've tested a few games and every one of them is stuck during the prefix creation. I can wait minutes with no activity.
This is with Experimental
```
GameAction [AppID 311340, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
Proton: Upgrading prefix from None to 8.0-102 (/mnt/NVME2/Gry/SteamLibrary/steamapps/compatdata/311340/)
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
```
This is with Proton 8.0-2
```
GameAction [AppID 311340, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
Proton: Upgrading prefix from None to 8.0-101 (/mnt/NVME2/Gry/SteamLibrary/steamapps/compatdata/311340/)
Proton: Error while copying to "/mnt/NVME2/Gry/SteamLibrary/steamapps/compatdata/311340/pfx/drive_c/windows/system32/d3d12core.dll": No such file or directory
Proton: Error while copying to "/mnt/NVME2/Gry/SteamLibrary/steamapps/compatdata/311340/pfx/drive_c/windows/syswow64/d3d12core.dll": No such file or directory
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
```

The games will successfully create a Prefix when using the Proton 7.0-6. They will also successfully upgrade, if the prefix was created in an earlier version. Another major annoyance is that even after cancelling the operation and closing Steam, the process will relaunch itself in the background with a winedevice, install script evaluator and a few other processes. No matter how many times I kill them, they return and require a reboot. Without it I can't launch any other Steam game.

My specs:

Fedora Linux 38 (6.3.8-200.fc38.x86_64)
Intel i5-10400
16GB RAM
Nvidia RTX 3060 (535.54.03)

[Steam PC System Information](https://gist.github.com/jkrhu/4a9b24f06f747b5b912b2317b9c4be85)

**Comment by Plarpoon** [comment]
I can confirm having the same issue, but this is valid only for certain games.

For example:
- [Titanfall ](https://steamdb.info/app/1454890/)
- [Deep Rock Galactic](https://steamdb.info/app/548430/)
both suffer from this issue, and as you have mentioned it's only happening if you run them the first time directly with Proton 8, that is not happening if you had previously successfully executed them with Proton 7 or prior.

Same issue is not present instead for such games like:
- [Orcs Must Die 3](https://steamdb.info/app/1522820/)
Which works without any issues at first execution directly with Proton 8.

Important to say that I had none of these issues when running these games directly under Proton 8 on the Steam Deck,these are issues I have always had exclusively with regular Linux on my PC.

This was tested on both Fedora 38 and NixOS.

This issue I had created I just discovered actually being related to this specific problem and not to only Titanfall.

https://github.com/ValveSoftware/Proton/issues/4410#issuecomment-1586400051

**Comment by jkrhu** [comment]
I would like to also point out that this problem is occurring only on the Steam client. I can still create new prefixes with new Wine and Proton versions from Bottles or Heroic Games Launcher. I would assume it's connected to some of the changes in the new client recently. I've been testing the beta some time ago and it was fine then, but not the more recent updates.

**Comment by DownloadFreeloader** [comment]
I believe I may be having a similar issue. Recently I have been trying Next Fest demos and they have all been following a pattern where they will not launch with Proton 8 or Experimental (will say launching, running, then instantly close), will fail to actually run with Proton 7 (will get stuck at either launching or running), but will then work normally after switching back to 8. Using the Thronefall Demo as an example, this is what was output by PROTON_LOG=1 on the initial launch with Proton 8: https://pastebin.com/qxhibq0z
When switching to Proton 7, this was the log output, with the last line I believe being caused by me clicking "Stop" in the Steam client: https://pastebin.com/nFqw9gCE

**Comment by Plarpoon** [comment]
Yes, that is 100% compatible with the same issues I was having too. Which sucks because I am not sure it will be an easy fix.

**Comment by jkrhu** [comment]
Today after the SELinux policy update on Fedora 38, Proton 8 and Experimental started working for me again.

**Comment by TheEpicNoobZilla** [comment]
Fedora 38 and still not working. Had to delete prefix made by Proton 8 (both Valve's and GE)/Experimental then launch the game with proton 7 or older to create prefix that actually launches the game. Proton log gives no information that would help since it fails to launch the game and despite steam detecting this state as in game

**Comment by ralgar** [comment]
I'm also suffering a similar issue on Arch Linux. The only difference seems to be that Proton 8.0 refuses to work at all - upgrading the prefix from Proton 7.0 doesn't work for me. It just hangs on "Launching" indefinitely. The various processes also won't terminate when stopping the game, shutting down Steam, or logging out - forcing me to reboot the system.

I had exactly the same log output as @jkrhu the other day, however with the new client update today I get more verbose output:

[steamlog.txt](https://github.com/ValveSoftware/Proton/files/12034000/steamlog.txt)

I also managed to get a Proton log today, indicating that it is at least trying to start the game process now:

[steam-782330.log](https://github.com/ValveSoftware/Proton/files/12033999/steam-782330.log)

@kisak-valve Can we get some eyes on this please? It's a pretty major issue. Like @jkrhu, I suspect this is caused by the new Steam client itself, as I had no issues with Proton 8.0-2 prior to the recent Steam client updates, nor have I ever had issues in the last 4 years I've been running this setup.

**Comment by Plarpoon** [comment]
I agree, this is definitely something major to look into. It's distro independent and seems to ubiquitously exist across all minor versions of Proton 8. As of now I had the issue in NixOS, Fedora 38, Fedora 39 and PopOS!

**Comment by jkrhu** [comment]
For me the problem stopped occuring after recent updates. I have a hunch it might be driver dependent too. I was initially on 535, but had problems with screen tearing. Eventually rolled back to 530, I kinda left it since there was no development on the issue. There were a bunch of updates like SELinux and when I tried a fresh prefix, it started working again. What drivers are you running? Have you tried with 530? I think the 535 has massive issues. The shader cache is broken, it has screen tearing on Wayland etc.

**Comment by alasky17** [comment]
Is anyone who is experiencing this issue *not* on NVIDIA with 535 drivers?  Someone who was experiencing something that sounded exactly like the issue described was able to narrow it down to the 535 drivers, and rolling back to 530 fixed the issue.  @jkrhu it sounds like at least in your case, this was the problem?

**Comment by Plarpoon** [comment]
Several of us have that issue on full AMD setups, so I am doubtful it is just a driver issue. Maybe some internal dependency.

**Comment by alasky17** [comment]
@Plarpoon Thank you.  I saw your post a while ago and tested Deep Rock Galactic locally, and could not repro following your steps :( (also with an all AMD setup with Arch).    To start with, could you copy your system info from your original comment on the Titanfall bug so that it is all in one place? 

Then, there are a couple of things for you to try: 
1) Please update to Mesa 23.1.3 if you are still on 23.1.1.
2) Please disable gamemode and manguhud entirely.
3) If you are still getting failures with a fresh prefix for any/all of the games mentioned after completing these steps, could you please get a log using "PROTON_LOG=+vulkan %command%" for each failing game?

**Comment by Plarpoon** [comment]
1) I am running always on the latest package updates, I am sort of an update freak :)
2) I did run without any parameter and it didn't change much unfortunately
3) [steam-548430.zip Deep Rock Galactic](https://github.com/ValveSoftware/Proton/files/12088198/steam-548430.zip)

I unfortunately don't have Titanfall 1 installed anymore as it was quite quick to complete and go to Titanfall 2. But I will post more updates whenever something shows up!

Thanks for the interest @alasky17, anything else you need just let me know and I will provide it. Even if it's boring to do, I just want to solve this issue!

P.S.
For the DRG log I uploaded I had re-enabled gamemode, I did test it without before but it still didn't work!

**Comment by alasky17** [comment]
@Plarpoon This is indeed boring :P , but could you please get a couple more logs?  1) It would be useful to get a log with gamemode disabled, just on the off chance that there are two failures going on, and disabling gamemode changes the failure.  2) Could you get a log of the game launching correctly on Proton 7.0-6 for comparrison?

**Comment by Plarpoon** [comment]
@alasky17


[DRG - Proton 7.0-6 - No extra parameters](https://www.mediafire.com/file/l4e8upmhpk1uqbs/steam-548430_-_Proton_7.zip/file) the game without gamemode is a lag fest when all in Ultra, absolutely terrible performance, but playable.

[DRG - Proton 8.0-2 - No extra parameters](https://github.com/ValveSoftware/Proton/files/12095796/steam-548430.log) unlike the version with 7.0-6 that has virtually no loading screen as they are extremely fast, the one with 8.0-2 actually takes approximately one entire minute after which it crashes to desktop (never managing to give me an actual playable character). After that for another minute Steam thinks I am actually still in-game despite the game being already crashed, it's only after this extra time that the option to "Play" the game comes available again! (so the process remains open despite the crash for some time)

These are the launch settings I used
![image](https://github.com/ValveSoftware/Proton/assets/40299833/b56aedcb-4f9b-4497-ba8a-9d76b780e0fb)

**Comment by alasky17** [comment]
@Plarpoon I think I'm a bit confused - I read your comment "both suffer from this issue, and as you have mentioned it's only happening if you run them the first time directly with Proton 8, that is not happening if you had previously successfully executed them with Proton 7 or prior." and took that to mean that the game was failing to launch when the prefix was created on Proton 8.0-x but the game works when the prefix is created on Proton 7.0-x and then upgraded.  Does creating the prefix on Proton 7.0-6 and then upgrading fix this issue with DRG?  Also, it sounds like the failure happens after a window is created?

**Comment by Plarpoon** [comment]
@alasky17 That's the odd thing, Titanfall 1 that I don't have installed anymore used to behave that way. If you created the symlinks with proton 7 then run it with 8 it would work, DRG on the other hand it works only with Proton 7.

And yes, I confirm that, the failure in Proton 8 happens well after the window is created. There is first a splashscreen and then a loading screen, in that loading screen it dies. On Titanfall it wouldn't even boot instead.

**Comment by runar-work** [comment]
I was also unable to reproduce any Deep Rock Galactic launch issues, but I can reproduce the Titanfall problem, so it seems like those are separate issues. Titanfall Proton log after a successful launch is absolutely massive, though.

- Kernel: 6.4.3-arch1-2
- GPU: RX 7600
- Driver: 23.1.3

**Comment by alasky17** [comment]
@Plarpoon It sounds like we should move the Deep Rock Galactic issue back to https://github.com/ValveSoftware/Proton/issues/284 because it is a game-specific issue, not a general problem, and it is not related to creating the prefix on Proton 7 vs Proton 8.  Could you please write a comment there with your system specs, the current behavior on Proton 7 and Proton 8, and the logs you collected without gamemode or manghud to make this easier to follow?

**Comment by runar-work** [comment]
I decided to test Monster Hunter Rise installation again, just to be absolutely certain it wasn't the same problem, and it turns out I get the same crash in the prefix installation even after I've switched to an AMD GPU. I just need to have the Nvidia driver installed, and the prefix installation will still fail with the same crash in `libnvidia-glcore.so.535.54.03`. After removing the driver and rebooting, it installed normally again.

For a brief while I could no longer reproduce the Titanfall issue. I deleted the prefix and it still launched normally with Proton 8.0. Now it's back to failing again. I collected various logs, mostly failures to launch and one from a successful 8.0 launch. The Proton log easily reaches 500 MB before the game even launces, so I just shut down the game via the Steam UI once it launched to keep it a bit shorter.

[Titanfall_launch_issues.zip](https://1drv.ms/u/s!Amz0nBA4PrbSm0b8QvvEKH4eXxSN?e=YWrQ7K)

**Comment by jkrhu** [comment]
@alasky17 Yes, I would say that the 535 series driver was the culprit in my case. Works fine on 530.

**Comment by alasky17** [comment]
@runar-work wrt the nvida-535 crash still happening on AMD as long as the library is installed; strange! Thank you for the update, and it sounds like we need to continue to investigate this further.   

@Plarpoon Thank you for the consolidated report on the DRG issue; greatly appreciated! 

@jkrhu Thank you for the confirmation :) 

---------------

@runar-work @Plarpoon wrt Titanfall - we are also looking into this now and can repro but no leads on a cause so far.  This seems like it is also game-specific and not related to the nvidia-535 library issue, so I think it will make more sense for future discussion to happen on the game thread following up on @Plarpoon's original comment here:  https://github.com/ValveSoftware/Proton/issues/4410#issuecomment-1586400051

----------

As far as I can tell, the original issue in this thread is the nvidia 535 library issue causing prefix creation failure, which is apparently not specific to NVIDIA GPUs but rather the library itself.    Any other issues that happen without the nvidia 535 library installed on a system seem to be game specific and are being shuttled to the game threads.  Since the NVIDIA driver issue has not been solved, I think it makes sense to keep this thread open to track that specifically.  

Thanks to everyone for the responsiveness to help to detangle this issue.  It is greatly appreciated :)

**Comment by Plarpoon** [comment]
Thanks to you man!

**Comment by jkrhu** [comment]
Problem is still occuring on the 535.86.05 driver

**Comment by reedlove** [comment]
Guys, I don't think that this is being caused by NVIDIA driver regressions. I too have noticed the stuttering and such when there used to be none.
What I believe is causing the prefix from being created for a number of these games is that DXSETUP.exe from the "Steamworks Common Redistributables" is hanging under Proton 8 during the first time setup of some of these games.

**Comment by reedlove** [comment]
Example using Proton 7.0-6:
$ export WINEPREFIX=~/test
$ cd ~/.steam/steam/steamapps/common/Proton 7.0/dist/bin
$ ./wine ~/.steam/steam/steamapps/common/Steamworks\ Shared/_CommonRedist/DirectX/Jun2010/DXSETUP.exe /silent

^ That installs the directx runtime in the prefix that I manually specified via WINEPREFIX. There are some typical "fixme" messages, but it installs correctly.

Now if I delete the prefix I created and try again using Proton 8.0-3:
$ rm -rf ~/test
$ cd ~/.steam/steam/steamapps/common/Proton 8.0/dist/bin
$ ./wine ~/.steam/steam/steamapps/common/Steamworks\ Shared/_CommonRedist/DirectX/Jun2010/DXSETUP.exe /silent

^ Installation freezes after outputting the following:
wine: created the configuration directory '/home/reed/test'
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
002c:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
004c:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
004c:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
004c:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
004c:err:ole:apartment_get_local_server_stream Failed: 0x80004002
0058:err:ole:com_get_class_object class {ff48dba4-60ef-4201-aa87-54103eef594e} not registered
0058:err:ole:com_get_class_object no class object {ff48dba4-60ef-4201-aa87-54103eef594e} could be created for context 0x1
0058:err:tabtip:create_uia_event_handler Failed to create IUIAutomation interface, hr 0x80040154
008c:fixme:file:NtLockFile I/O completion on lock not implemented yet
008c:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:err:mscoree:LoadLibraryShim error reading registry key for installroot
008c:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 1)
008c:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 1)
00e8:fixme:msg:pack_message msg 14 (WM_ERASEBKGND) not supported yet
0110:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0110:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0110:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0110:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0110:fixme:wineusb:query_id Unhandled ID query type 0x5.
wine: configuration in L"/home/reed/test" has been updated.
0184:fixme:oleacc:find_class_data unhandled window class: L"#32769"
0184:fixme:uiautomation:uia_get_providers_for_hwnd Override provider callback currently unimplemented.
0184:fixme:uiautomation:msaa_provider_GetPropertyValue Unimplemented propertyId 30024
0184:fixme:uiautomation:msaa_fragment_get_FragmentRoot 00000000008FAEA8, 000000000021FA90: stub!
0190:fixme:uiautomation:uia_get_providers_for_hwnd Override provider callback currently unimplemented.
0190:fixme:uiautomation:msaa_provider_GetPropertyValue Unimplemented propertyId 30024
0190:fixme:uiautomation:msaa_fragment_get_FragmentRoot 0000000000928CA8, 0000000001B2FAB0: stub!

**Comment by ivyl** [comment]
Nvidia has been notified of this on 14th of July.

It looks like some regression in their 32bit driver. When DXSETUP.EXE is being run it crashes in the Nvidia driver (/usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu/libnvidia-glcore.so.535.54.03 at offset 0x1739e1d) on the Unix side. It started happening with this driver version specifically and it still works okay with an older one.

It crashes when we are calling [`vkCreateInstance()`](https://github.com/ValveSoftware/wine/blob/8a8ec5f86d8ab1e1d4c6bc88dda016b5e8cf479e/dlls/winex11.drv/xrandr.c#L754) in winex11.drv. Things that get passed via all the Wine layers to the actual driver look correct. Vulkan Validation Layers also did not complain.

**I wasn't able to repro with 535.86.05 though.** Does the issue still persist for you folks?

**Comment by reedlove** [comment]
@ivyl Are you able to explain why it works in Proton 7 and not 8?
Using the 535.54.03 drivers, that is.

**Comment by runar-work** [comment]
Yeah, I still have the same problem. Here's what I got with Vampyr and Proton 8.0 with 535.86.05:
```
24371.668:0324:0328:warn:seh:handle_syscall_fault backtrace: --- Exception 0xc0000005 at 0x6f26442d: /usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu/libnvidia-glcore.so.535.86.05 + 0x173b42d.
```

**Comment by ivyl** [comment]
> @ivyl Are you able to explain why it works in Proton 7 and not 8? Using the 535.54.03 drivers, that is.

No idea. Not much has changed on the Wine side and I don't know what the driver is doing, it's proprietary.

Since the older drivers were working just fine with 8.0 for months, other vendors do not have the same problem and validation layers do not complain it seems more productive to focus on why the driver has regressed.

**Comment by jkrhu** [comment]
> Does the issue still persist for you folks?

Yes, it does. Still requires the Proton 7 workaround during prefix creation. Upgrading works fine. The 535 series is very unfortunate. Major regressions in different spots. Hope it gets sorted out soon.

**Comment by ralgar** [comment]
@ivyl Can confirm the issues persist for me, as of last night. Thanks for confirming the driver as the cause.

**Comment by adamnv** [comment]
Can someone facing this issue please post the output of running '`getenforce`' or '`grep SELINUX= /etc/selinux/*|grep -v #`'?  Thanks!

**Comment by reedlove** [comment]
> 

Actually, it returns nothing.

The only uncommented lines are:

module-store = direct
expand-check=0

On Linux Mint 21.2

**Comment by adamnv** [comment]
Thanks, @reedlove - I assume you don't have `getenforce`?  To the best of your knowledge you're not running with selinux enabled? :)  That'd rule that out, then...

**Comment by reedlove** [comment]
> Thanks, @reedlove - I assume you don't have `getenforce`? To the best of your knowledge you're not running with selinux enabled? :) That'd rule that out, then...

Correct, selinux is not enabled. XD hah

**Comment by Poslovitch** [comment]
> Still requires the Proton 7 workaround during prefix creation

The workaround doesn't work on my end : "compatibility tool failed" with Proton 7-6.

**Comment by jkrhu** [comment]
> > Still requires the Proton 7 workaround during prefix creation
> 
> The workaround doesn't work on my end : "compatibility tool failed" with Proton 7-6.

Did you remove the compatdata game subfolder before forcing Proton 7.0-6?

**Comment by Poslovitch** [comment]
Yup. But Proton doesn't even try to create a prefix: it seems to fail before that. At least, I don't see Proton talking about prefixes in the logs.

And there's the log when trying to load the game with Proton 8 (after trying to launch it with 7):
```
Proton: Upgrading prefix from None to GE-Proton8-9 (/software/steam/steamapps/compatdata/2239150/)
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
```

Okay, err... Did Ctrl+C and here's what it printed out:

```
^Csrc/clientdll/installscript_posix.cpp (531) : Assertion Failed: Standalone evaluator returned error code for app 2239150
assert_20230803211806_37.dmp[41149]: Uploading dump (out-of-process)
/tmp/dumps/assert_20230803211806_37.dmp
/bin/sh\0-c\0PROTON_LOG=1 /home/poslovitch/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=2239150 -- /home/poslovitch/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/home/poslovitch/.steam/debian-installation/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/poslovitch/.steam/debian-installation/compatibilitytools.d/GE-Proton8-9'/proton waitforexitandrun  '/software/steam/steamapps/common/Thronefall/Thronefall.exe'\0
chdir "/software/steam/steamapps/common/Thronefall"
ERROR: ld.so: object '/home/poslovitch/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/poslovitch/.steam/debian-installation/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/poslovitch/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/poslovitch/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/poslovitch/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
WARNING: discarding _NET_WM_PID 2 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 2 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 2 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 2 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
assert_20230803211806_37.dmp[41149]: Finished uploading minidump (out-of-process): success = yes
assert_20230803211806_37.dmp[41149]: response: CrashID=bp-006665b8-c2f5-4fe5-aff0-edb6d2230803
assert_20230803211806_37.dmp[41149]: file ''/tmp/dumps/assert_20230803211806_37.dmp'', upload yes: ''CrashID=bp-006665b8-c2f5-4fe5-aff0-edb6d2230803''
ProtonFixes[41274] INFO: Running protonfixes
ProtonFixes[41274] INFO: Running checks
ProtonFixes[41274] INFO: All checks successful
ProtonFixes[41274] INFO: Using global defaults for "Thronefall" (2239150)
ProtonFixes[41274] INFO: No protonfix found for "Thronefall" (2239150)
```

It's still not running, but Steam shows it as such (which is somehow better than it being stuck on "Launching...", I guess?).

**Comment by jkrhu** [comment]
The issue is fixed on Nvidia 535.98 driver. Games which wouldn't create a prefix under Proton 8 previously on the 535 series, do work correctly now on my system.

```
Proton: Upgrading prefix from None to 8.0-104 (/mnt/NVME/Games/Steam Library/steamapps/compatdata/311340/)
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
/bin/sh\0-c\0PROTON_LOG=1 /home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=311340 -- /home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton - Experimental'/proton waitforexitandrun  '/mnt/NVME/Games/Steam Library/steamapps/common/Metal Gear Solid Ground Zeroes/MgsGroundZeroes.exe'\0
chdir "/mnt/NVME/Games/Steam Library/steamapps/common/Metal Gear Solid Ground Zeroes"
ERROR: ld.so: object '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jkrhu/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
08/11 22:45:21 Init: Installing breakpad exception handler for appid(gameoverlayui)/version(20230728184426)/tid(1924)
                                     08/11 22:45:21 Init: Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)/tid(1924)
                                                               Game 311340 created interface STEAMAPPLIST_INTERFACE_VERSION001 / AppList
Game 311340 created interface STEAMAPPS_INTERFACE_VERSION006 / Apps
Game 311340 created interface STEAMAPPTICKET_INTERFACE_VERSION001 / 
Game 311340 created interface STEAMCONTROLLER_INTERFACE_VERSION / Controller
Game 311340 created interface STEAMHTMLSURFACE_INTERFACE_VERSION_002 / HTMLSurface
Game 311340 created interface STEAMHTTP_INTERFACE_VERSION002 / HTTP
Game 311340 created interface STEAMMUSICREMOTE_INTERFACE_VERSION001 / MusicRemote
Game 311340 created interface STEAMMUSIC_INTERFACE_VERSION001 / Music
Game 311340 created interface STEAMREMOTESTORAGE_INTERFACE_VERSION012 / RemoteStorage
Game 311340 created interface STEAMSCREENSHOTS_INTERFACE_VERSION002 / Screenshots
Game 311340 created interface STEAMUGC_INTERFACE_VERSION003 / UGC
Game 311340 created interface STEAMUNIFIEDMESSAGES_INTERFACE_VERSION001 / UnifiedMessages
Game 311340 created interface STEAMUSERSTATS_INTERFACE_VERSION011 / UserStats
Game 311340 created interface SteamFriends014 / Friends
Game 311340 created interface SteamMatchMaking009 / Matchmaking
Game 311340 created interface SteamMatchMakingServers002 / MatchmakingServers
Game 311340 created interface SteamNetworking005 / Networking
Game 311340 created interface SteamUser017 / User
Game 311340 created interface SteamUtils007 / Utils
Game 311340 method call count for IClientRemoteStorage::FileExists : 6
Game 311340 method call count for IClientRemoteStorage::GetFileTimestamp : 6
Game 311340 method call count for IClientRemoteStorage::GetFileSize : 2
Game 311340 method call count for IClientRemoteStorage::FileRead : 1
Game 311340 method call count for IClientRemoteStorage::FileWrite : 2
Game 311340 method call count for IClientUserStats::GetAchievementDisplayAttribute : 2
Game 311340 method call count for IClientUserStats::GetAchievement : 1
Game 311340 method call count for IClientUserStats::RequestCurrentStats : 1
Game 311340 method call count for IClientUtils::RecordSteamInterfaceCreation : 22
Game 311340 method call count for IClientUtils::GetAppID : 26
Game 311340 method call count for IClientUser::BIsSubscribedApp : 12
Game 311340 method call count for IClientUser::GetAppOwnershipTicketExtendedData : 1
Game 311340 method call count for IClientUser::GetUserDataFolder : 6
Game 311340 method call count for IClientUser::GetSteamID : 4
Game 311340 method call count for IClientUser::BLoggedOn : 2
Uploaded AppInterfaceStats to Steam 
```

**Comment by reedlove** [comment]
Just tried installing Tekken 7 using Proton 8 with nvidia-driver-535/jammy,now 535.98-0ubuntu0~gpu22.04.1 from the Ubuntu graphics team PPA and DXSETUP.exe is still freezing at first install.

**Comment by reedlove** [comment]
So explain this...
If I go in to Downloads and clear the cache and then delete the compatdata folder and then run the game for the first time, DXSETUP.exe works just fine.
Explain to me again how this is an nvidia problem and not a problem with Proton 8.0-3 or Steam?

**Comment by jkrhu** [comment]
In my own experience I've already installed a bunch of titles like MGS:GZ, Forspoken, WD2 and RE7. Every one of them successfully created a prefix after fresh install. I'm not sure why yours failed, perhaps some leftover files in compatdata from previous installations/prefixes. Could be also some incompatibility with trying to upgrade Proton 7 prefixes created using the previous 535 drivers.

**Comment by reedlove** [comment]
So now get this. AFTER I run Tekken 7, "Steamworks Common Redistributables" has an update. I update it.
I go to run Overcooked! for the first time on this fresh install of Steam, using Proton 8.0-3.
Guess what's hung again..... DXSETUP.exe

**Comment by reedlove** [comment]
> In my own experience I've already installed a bunch of titles like MGS:GZ, Forspoken, WD2 and RE7. Every one of them successfully created a prefix after fresh install. I'm not sure why yours failed, perhaps some leftover files in compatdata from previous installations/prefixes. Could be also some incompatibility with trying to upgrade Proton 7 prefixes created using the previous 535 drivers.

This was a "first time" install of Tekken 7 on a fresh install of Steam on a fresh install of Linux Mint 21.2.

**Comment by reedlove** [comment]
Just cleared out the entire compatdata folder and reinstalled a number of games from scratch and everything appears to be working now. -_-
Wacky.

**Comment by ivyl** [comment]
Good news - there's some progress. The issue seems to be due to address space exhaustion. DXSETUP is a 32 bit process and it can address 4G of memory. Nvidia's driver starting with 535 seem to be a bit more heavy than the previous series. While the bump is not colossal it's enough to make things crash. Nvidia is aware of this.

**One thing that should help with the issue is uninstalling all the other Vulkan drivers that are not in use. So if you are on Nvidia uninstall all things Mesa, including extra Vulkan layers, and vice versa.** Otherwise they are all loaded using the precious address space.

That said I'm looking into where the memory goes in places we have control over it and why DXSETUP is this pathological.

**Comment by reedlove** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6859#issuecomment-1682357594

Yay! An explanation that appears to make sense! Good deal! Thanks for your hard work!

**Comment by pchome** [comment]
@ivyl 
What if you just disable Vulkan layers for the setup process?

Use `VK_LOADER_DEBUG=layer` (or =driver, =debug, =info, =all) to see what's going on.
To disable known unwanted layers (steam layers (?), was enabled by steam client (?)) either set "DISABLE" environment variables for the process or blacklist them via a meta layer.

<details><summary>Meta layer example</summary>

Created by old `vkconfig` utility when on/off selected layers.
`~/.local/share/vulkan/implicit_layer.d/VkLayer_override.json`
```
{
    "file_format_version": "1.1.2",
    "layer": {
        "api_version": "1.3.211",
        "blacklisted_layers": [
            "VK_LAYER_VALVE_steam_fossilize_32",
            "VK_LAYER_VALVE_steam_fossilize_64",
            "VK_LAYER_NV_optimus",
            "VK_LAYER_VALVE_steam_overlay_32",
            "VK_LAYER_VALVE_steam_overlay_64",
            "VK_LAYER_MESA_device_select"
        ],
        "component_layers": [
        ],
        "description": "LunarG Override Layer",
        "disable_environment": {
            "DISABLE_VK_LAYER_LUNARG_override": "1"
        },
        "implementation_version": "1",
        "name": "VK_LAYER_LUNARG_override",
        "override_paths": [
        ],
        "type": "GLOBAL"
    }
}
```
This is **implicit** layer w/o `enable_environment` section so will be auto-loaded. **Add "ENABLE" variable to load it only when needed** (`enable_environment` section).

</details>

**Comment by EnLaLuna91** [comment]
Hello,

The same thing happens to me with Baldurs Gate 3. I've tried with Proton Experimental, Proton 8.0-3 and Proton GE.

[Steam log](https://gist.github.com/EnLaLuna91/27ae45abc5aca90512d0897f6b085890)

[Harware info](https://gist.github.com/EnLaLuna91/155e7d398499dc2b8e586866de3c4391)

I had been playing for days, until the Proton GE version was updated and since then I can't play, independently of the Proton version.

**Comment by reedlove** [comment]
Any news on this front?

**Comment by ArcadesSaboth** [comment]
I tried to log what layers I'm using as suggested by @pchome and I tried disabling them all, e.g.:
```
DISABLE_VK_LAYER_VALVE_steam_overlay_1=1 NODEVICE_SELECT=1 DISABLE_MANGOHUD=1 VK_LOADER_DEBUG=layer steam
``` 
So I ended up with no layers at all and I still can't create a prefix. I tested with **Trover Saves the World**, a UE4 game I happened to purchase yesterday and having this problem.

I would like to note that I am having a similar problem with **The Elders Scrolls Online** where the prefix is created but the installation of the supplied Microsoft VC++ 2010 hangs for no apparent reason on Proton 8 but works on Proton 7. I'm not sure if this problem is related to this problem. Additional note: for this game it hangs even if I try to install it with Protontricks.

I'm on Fedora 38, Nvidia drivers 535.98.

**Comment by pchome** [comment]
Still guessing, but as temporary solution people could try to rename the installscript.vdf file inside the game's directory (the one contains DXSETUP).
In most cases those installations are not required on linux.
Or just copy `d3dx*.dll` files from the other `compatdata` prefix directory (`pfx/drive_c/windows/{system32,syswow64}/`).

Or another variant: copy missed dlls into e.g. `/opt/dlls/{x32,x64}/` and use  `WINEPATH="Z:\\opt\\dlls\\x32;Z:\\opt\\dlls\\x64"`. But this will work only for missing files, or otherwise wine's `d3dx*.dll` will be used.

I guess from proton's side there could be an option to ignore installation process, or maybe they could bundle d3dx* and use the WINEPATH trick.

**Comment by reedlove** [comment]
Still no update? Not being able to reliably create a prefix using the latest version of Proton seems like a pretty big deal. I know that everything wrong under the sun with Linux graphics is NVIDIA's fault and all (according to everyone), but somehow many games and other pieces of software still seem to work just fine using their proprietary drivers.

**Comment by ralgar** [comment]
@reedlove Valve neither has control over DXSETUP nor the Nvidia driver, so it would seem likely there isn't a lot they can do on this one.

I doubt Microsoft will lift a damn finger unless this affects Windows as well, so it's basically up to Nvidia to trim their driver back down, and with their pace of development on Linux who knows how long that will take.

This is the unfortunate downside of translation layers - we are still beholden to Microsoft and their ecosystem in some ways.

---

I do wonder if a LAA patch, like we used to apply to 32 bit Skyrim, might keep it from crashing. Has anybody looked into such a thing?

I'm not especially hopeful, but I might try it this weekend if I get a chance.

**Comment by kisak-valve** [comment]
`PROTON_FORCE_LARGE_ADDRESS_AWARE` has been enabled by default for a fair while now, and should apply to everything 32 bit.

**Comment by ralgar** [comment]
@kisak-valve Had a feeling y'all would've done something to this effect. Thanks for confirming :)

**Comment by reedlove** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6859#issuecomment-1710939726

I'm aware and am being a bit sarcastic. ;]
Also, I thought that large address awareness pertained to system memory addressing limitations and not necessarily gpu memory addressing as I thought it would be handled via the driver and graphics API.

**Comment by ralgar** [comment]
@reedlove Gotcha. Sometimes it's difficult to tell in text :)

Unless I'm grossly misunderstanding the issue, as laid out above by Ivy, this *is* related to DXSETUP exhausting its addressable system memory. You are otherwise correct in your assertion though.

**Comment by liberodark** [comment]
Hi, 
hope we find solution about this.

Best Regards

**Comment by ElmerFuddDK** [comment]
I too am experiencing issues with proton not creating prefixes.
I'm on MX-Linux (debian 12 based) on an AMD based SoC, so fully mesa.

I tried exporting various combos of the following before starting:
export VK_ICD_FILENAMES=
export PROTON_FORCE_LARGE_ADDRESS_AWARE=1
export PROTON_LARGE_ADDRESS_AWARE=1
export WINE_LARGE_ADDRESS_AWARE=1

Each time the prefix creation process hang with the process:
C:\windows\syswow64\rundll32.exe setupapi,InstallHinfSection Wow64Install 128 \\?\Z:\...\usr\share\wine\wine.inf
Killing this would exit the startup with a failure:
wine: could not load kernel32.dll, status c0000135

I tried with proton 8.0-c and experimental-8.0-20230907

Wine 8.12-64bit-staging works fine. Proton 7 works fine.

**Comment by ivyl** [comment]
Current Nvidia beta driver 535.43.09 seems to help with the memory usage significantly. Me and other people who had local repro are no longer able to hit the problem even when having both Mesa and Nvidia blobs installed and loaded into the address space.

We are still looking for potential address space savings inside Wine / Proton but it's unlikely that we'll find anything significant.

FWIW there's upstream work to have working wow64 in Wine - the Windows part will run in 32bits but the Linux side will be full on 64bits. There's still ways to go to have it complete and working well, we are talking multiple Proton rebases away.

@ElmerFuddDK Please verify your proton installation. You issue seems unrelated to the OOM problem. ` wine: could not load kernel32.dll, status c0000135` means the DLL was not found. That's usually a symptom of busted Proton installation.

**Comment by jkrhu** [comment]
For me, I had no problems with it since 535.98. I'm on default Fedora 38, 535.104.05 drivers and it's solid on my 3060. Grub settings disable nouveau by default and I also enable the nvidia-drm.modeset. I still have mesa packages installed. Not sure why it only works for some users now.

**Comment by ElmerFuddDK** [comment]
@ivyl that error is shown when I kill the hanging process, and then proton exits.
I just did a re-test and proton can create prefixes from directly from steam now. I'm running proton "command line", either from  the installed proton or via proton-GE in lutris.
You may be right that it is not related to this issue. It's might be an odd coincidence it hangs in syswow64 initialization.

**Comment by jinzhongjia** [comment]
now, I also meet this problem on archlinux, and my laptop has an Intel CPU and NVIDIA graphics card
on Deep Rock Galactic, I must first create the prefix through proton7 and then start it through proton8

**Comment by pellcorp** [comment]
Can confirm rolling back to NVIDIA 525 drivers on Ubuntu 22.04 fixes the dxsetup issue.

Previously on 535 I would get the error for most games I would try to install, seems like perhaps there might be timing issues here too, but now with 525 it's consistently installing games again, which is a relief as I reinstalled my sons Ubuntu and steam and wanted to re-download all his big games so it would be ready for him when he got back, having to explain that you need to wait say 10 minutes and then kill the dxsetup process from monitoring would have got old real quick

Hopefully a better solution with 535 drivers will be provided

**Comment by Jan200101** [comment]
Also managed to recreate this with Titanfall 2 and the EA Desktop Installer

**Comment by psycowithespn** [comment]
I believe I ran into this on NixOS Unstable, Nvidia 980ti. Game was Killing Floor 2. I was using nixpkgs#steam.

However, I believe I did NOT run into this when I tried Flatpak steam with the same game, same Proton version, and same version of nixpkgs. Is there anything specific I can capture to help with this?

**Comment by jinzhongjia** [comment]
Ultra Street Fighter IV can also meet this problem

**Comment by chiat88** [comment]
Glad this thread is active.
Valve support mentioned they only support Ubuntu among all linux distros.

Fedora 37 + nvidia user here.
Proton used: 8.0-4, and Proton Experimental
nvidia driver: 535

Just to add log on
Octopath Traveler 2 stuck at DXSETUP.exe. After manually killing it, the game will launch.
I am expecting launcher should be able to bypass DXSETUP.

It frustrated me as it won't produce $PROTON_LOG at all (cause it is not even starting).
*Note: Workaround on clearing download cache worked 1 time or 2, problem will repeat again.

**Comment by ralgar** [comment]
Just chiming in again to confirm that this appears to finally be resolved with driver 545.

**Comment by 4nexus5** [comment]
Any new updates or solutions? I am on AMD GPU and the issue still persists. None of the solutions I found online worked for me (disabling shader pre-caching, reinstalling, downloading additional drivers etc.). I am on Arch Linux

**Comment by jinzhongjia** [comment]
This problem seems not happen on myself now. Maybe you can try proton9-ge? or proton expermental ?

**Comment by 4nexus5** [comment]
> This problem seems not happen on myself now. Maybe you can try proton9-ge? or proton expermental ?

Nope, switched between multiple Proton versions, the issue still persists

**Comment by jkrhu** [comment]
> > This problem seems not happen on myself now. Maybe you can try proton9-ge? or proton expermental ?
> 
> Nope, switched between multiple Proton versions, the issue still persists

This is a different issue. The topic is about the Nvidia propietary driver and DirectX redistributable causing exhaustion of 32bit memory address space. Which prevented the prefix from being created. 

I would suggest to open a new issue if you have similar issues on RADV. The Nvidia issue has been solved since then. 

@kisak-valve @ivyl Should we close this Nvidia issue?

**Comment by ninetailedtori** [comment]
I've got this issue...on experimental? And 8. But not 9. With Nvidia GPU drivers, 565. Wayland. If relevant, using protontricks through experimental, I see a kernel32.dll write error as well, so I'm not sure if it's the same or different issue.

**Comment by ninetailedtori** [comment]
nvm, my issue was clamav being triggery frickin happy...
